rs/rosetta-api/icp_ledger/ledger/tests/tests.rs

```diff
@@ -190,6 +192,108 @@ fn test_minting_account() {
+    assert_eq!(0u64, balance_of(&env, canister_id, anon.0));
+
+    // Transfer to the account of the anonymous principal using `icrc1_transfer` succeeds
+    // The expected block index after the transfer is 1 (0 is the initial mint to `p1`).
+    let mut expected_block_index = 1u64;
+    assert_eq!(
+        transfer(&env, canister_id, p1.0, anon.0, TRANSFER_AMOUNT).expect("transfer failed"),
+        expected_block_index
+    );
+    expected_block_index += 1;
+
+    // Transfer to the account of the anonymous principal using the ICP-specific `transfer` succeeds
+    let transfer_args = icp_ledger::TransferArgs {
+        memo: icp_ledger::Memo(0u64),
+        amount: Tokens::from_e8s(TRANSFER_AMOUNT),
+        fee: Tokens::from_e8s(FEE),
+        from_subaccount: None,
+        to: AccountIdentifier::new(anon, None).to_address(),
+        created_at_time: None,
+    };
+    let response = env.execute_ingress_as(
+        p1,
+        canister_id,
+        "transfer",
+        Encode!(&transfer_args).unwrap(),
+    );
+    let result = Decode!(
+        &response
+        .expect("failed to transfer funds")
+        .bytes(),
+        Result<BlockIndex, TransferError>
+    )
+    .expect("failed to decode transfer response");
+    assert_eq!(result, Ok(expected_block_index));
+
+    // Transfer from the account of the anonymous principal using `icrc1_transfer` fails
+    let transfer_arg = TransferArg {
+        from_subaccount: None,
+        to: Account::from(p1.0),
+        fee: None,
+        created_at_time: None,
+        amount: Nat::from(TRANSFER_AMOUNT),
+        memo: None,
+    };
+    let encoded_transfer_result = env
+        .execute_ingress_as(
+            anon,
+            canister_id,
+            "icrc1_transfer",
+            Encode!(&transfer_arg).unwrap(),
+        )
+        .expect("failed to transfer funds")
+        .bytes();
+    let string_from_bytes_result = String::from_utf8(encoded_transfer_result.clone());
+    assert_eq!(
+        string_from_bytes_result,
+        Ok("Anonymous principal cannot hold tokens on the ledger.".to_string())
+    );
+
+    // Transfer from the account of the anonymous principal using the ICP-specific `transfer` fails
+    let transfer_args = icp_ledger::TransferArgs {
+        memo: icp_ledger::Memo(0u64),
+        amount: Tokens::from_e8s(TRANSFER_AMOUNT),
+        fee: Tokens::from_e8s(FEE),
+        from_subaccount: None,
+        to: AccountIdentifier::new(p1, None).to_address(),
+        created_at_time: None,
+    };
+    let response = env.execute_ingress_as(
+        anon,
+        canister_id,
+        "transfer",
+        Encode!(&transfer_args).unwrap(),
+    );
+    assert!(response.is_err());
+    if let Err(err) = response {
+        assert_eq!(err.code(), ErrorCode::CanisterCalledTrap);
+        assert!(err.description().contains("Canister called `ic0.trap` with message: Panicked at 'Sending from 2vxsx-fae is not allowed'"));
```

Comment:
Maybe also worth adding a check that approve by anonymous principal fails.

Reply:
Done!